src/extensions/default/ImageUrlCodeHints/main.js

```diff
@@ -0,0 +1,814 @@
+
+            directory.getContents(function (err, contents) {
+                var currentDeferred, entryStr, syncResults;
+
+                if (!err) {
+                    contents.forEach(function (entry) {
+                        if (ProjectManager.shouldShow(entry)) {
+                            // convert to doc relative path
+                            entryStr = queryDir + entry._name;
+                            if (entry._isDirectory) {
+                                entryStr += "/";
+                            }
+
+                            // code hints show the unencoded string so the
+                            // choices are easier to read.  The encoded string
+                            // will still be inserted into the editor.
+                            unfiltered.push(entryStr);
+                        }
+                    });
+
+                    self.cachedHints.unfiltered = unfiltered;
+                    self.cachedHints.query      = query;
+                    self.cachedHints.queryDir   = queryDir;
+                    self.cachedHints.docDir     = docDir;
+
+                    if (self.cachedHints.deferred.state() !== "rejected") {
+                        currentDeferred = self.cachedHints.deferred;
+
+                        // Since we've cached the results, the next call to _getUrlList should be synchronous.
+                        // If it isn't, we've got a problem and should reject both the current deferred
+                        // and any new deferred that got created on the call.
+                        syncResults = self._getUrlList(query);
+                        if (syncResults instanceof Array) {
+                            currentDeferred.resolveWith(self, [syncResults]);
+                        } else {
+                            if (currentDeferred && currentDeferred.state() === "pending") {
+                                currentDeferred.reject();
+                            }
+
+                            if (self.cachedHints.deferred &&
+                                    self.cachedHints.deferred.state() === "pending") {
+                                self.cachedHints.deferred.reject();
+                                self.cachedHints.deferred = null;
+                            }
+                        }
+                    }
+                }
+            });
+
+            return self.cachedHints.deferred;
+        }
+
+        // build list
+
+        // without these entries, typing "../" will not display entries for containing folder
+        if (queryUrl.filename === ".") {
+            result.push(queryDir + ".");
+        } else if (queryUrl.filename === "..") {
+            result.push(queryDir + "..");
+        }
+
+        // add file/folder entries
+        unfiltered.forEach(function (item) {
+            result.push(item);
+        });
+
+        // TODO: filter by desired file type based on tag, type attr, etc.
```

Comment:
Let's use the `LanguageManager` module here to help us get the proper extensions for image file types + svg.  See my patch here for some ideas https://github.com/adobe/brackets/pull/10788/files#diff-a13729ff2e1bec651c063696caa00834R473